source/test/TestApp/PlayFabGroupsTest.cpp

```diff
@@ -0,0 +1,87 @@
+
+#include "TestAppPch.h"
+
+#include "PlayFabGroupsTest.h"
+#include "playfab/PlayFabApiSettings.h"
+#include "playfab/PlayFabSettings.h"
+
+using namespace PlayFab;
+
+namespace PlayFabUnit
+{
+
+    // GROUPS API
+    void PlayFabGroupsTest::GroupsApiTest(TestContext& testContext)
+    {
+        auto req = ClientModels::LoginWithCustomIDRequest();
+        req.CustomId = PlayFabSettings::buildIdentifier;
+
+        clientApi->LoginWithCustomID(req, Callback(&PlayFabGroupsTest::GroupsTestLoginCallback), Callback(&PlayFabGroupsTest::GroupsTestSharedFailureCallback), &testContext);
+    }
+
+    void PlayFabGroupsTest::GroupsTestLoginCallback(const ClientModels::LoginResult& result, void* customData)
+    {
+        auto groupApi = std::make_shared<PlayFabGroupsInstanceAPI>(result.authenticationContext);
+        auto req = GroupsModels::CreateGroupRequest();
+        req.CustomTags = std::map<std::string, std::string>();
+
+        // TODO Bug 29786037: this map is required to be filled to not get a 500 error with the CreateGroup api call
+        req.CustomTags.insert(std::pair<std::string, std::string>("One", "Two"));
+
+        req.GroupName = GenerateRandomString();
+        groupApi->CreateGroup(req, Callback(&PlayFabGroupsTest::GroupsTestGroupCallback), Callback(&PlayFabGroupsTest::GroupsTestSharedFailureCallback), customData);
+    }
+
+    void PlayFabGroupsTest::GroupsTestGroupCallback(const GroupsModels::CreateGroupResponse&, void* customData)
+    {
+        TestContext* testContext = static_cast<TestContext*>(customData);
+        testContext->Pass();
+    }
+
+    void PlayFabGroupsTest::GroupsTestSharedFailureCallback(const PlayFabError& error, void* customData)
+    {
+        TestContext* testContext = static_cast<TestContext*>(customData);
+        testContext->Fail("Expected call to succeed but instead got the error: " + error.GenerateErrorReport());
+    }
+
+    void PlayFabGroupsTest::AddTests()
+    {
+        AddTest("CreateGroupApi", &PlayFabGroupsTest::GroupsApiTest);
+    }
+
+    void PlayFabGroupsTest::ClassSetUp()
+    {
+        groupsTestSettings = std::make_shared<PlayFabApiSettings>();
+        groupsTestSettings->titleId = testTitleData.titleId;
+
+        clientApi = std::make_shared<PlayFabClientInstanceAPI>(groupsTestSettings);
+
+        // Bug 45669 - this is required for groups tests to pass.
```

Comment:
Why is this a bug?

Reply:
without that map, there is an issue on the server that returns a 500 error (not an "invalid parameter", it is a separate issue that this test is supposed to help isolate for future investigation.

Reply:
We should be able to run these tests with local values, but attempting to do so ran into odd timeout issues that I wanted to investigate at a later time. It seems like a threading issue since debugging will pass the test, but running without stepping through times the test out. It looks very similar to an issue I was seeing with QoS, but it is out of scope to this particular PR.